### DISCUSSION TITLE
Compile error using encoder on Pi Pico

### DISCUSSION THREAD

**Discussion by w9ran** [tttapa/Control-Surface#994] [Q&A]
Surprised to get this error when compiling the CCRotaryEncoder example.

Since Raspberry Pi Pico is shown as a supported platform, I'd appreciate any suggested workaround for the encoder interrupts. (Is the "kludge" recommended?

Thanks. Here is the complete error message string:

In file included from C:\Users\ranic\Documents\Arduino\libraries\Encoder/Encoder.h:46:0,
                 from C:\Users\ranic\Downloads\Control-Surface-main\Control-Surface-main\examples.old\medium-old\MIDIOutput\Control-Change\CCRotaryEncoder\CCRotaryEncoder.ino:29:
C:\Users\ranic\Documents\Arduino\libraries\Encoder/utility/interrupt_pins.h:334:2: error: #error "Interrupts are unknown for this board, please add to this code"
 #error "Interrupts are unknown for this board, please add to this code"
  ^~~~~
C:\Users\ranic\Documents\Arduino\libraries\Encoder/utility/interrupt_pins.h:337:2: error: #error "Encoder requires interrupt pins, but this board does not have any :("
 #error "Encoder requires interrupt pins, but this board does not have any :("
  ^~~~~
C:\Users\ranic\Documents\Arduino\libraries\Encoder/utility/interrupt_pins.h:338:2: error: #error "You could try defining ENCODER_DO_NOT_USE_INTERRUPTS as a kludge."
 #error "You could try defining ENCODER_DO_NOT_USE_INTERRUPTS as a kludge."
  ^~~~~
Using library Encoder at version 1.4.2 in folder: C:\Users\ranic\Documents\Arduino\libraries\Encoder 
Using library Control-Surface-main at version 2.0.0 in folder: C:\Users\ranic\Documents\Arduino\libraries\Control-Surface-main 
Using library SPI in folder: C:\Users\ranic\AppData\Local\Arduino15\packages\arduino\hardware\mbed_rp2040\3.1.1\libraries\SPI (legacy)
exit status 1

**Comment by tttapa** [comment]
You seem to be using an old, unsupported example. See https://tttapa.github.io/Control-Surface-doc/Doxygen/d2/d1f/RotaryEncoder_8ino-example.html instead.

The Pico is not supported by the Encoder.h library (which is where the error originates, not inside of Control Surface itself), but it is supported by the included encoder class used by the current examples.

**Comment by w9ran** [comment]
Indeed I was, unknowingly but thank you for the quick reply.  The current example compiles just fine.    Thank you also for this fantastic library!   Bob
